### ISSUE TITLE
Child porn, rape, murder ...

### ISSUE THREAD

**Issue by IGLOU-EU** [HelloZeroNet/ZeroNet#2746]
This is a very good and awesome tech, with a cool hosting feature.

But from my first try of Tor 15y ago and after testing this solution I will not use it again... 

No censoring at all, is not a good idea.

When I have tested this solution, I was exposed to unacceptable content. 

Child porn, rape, murder ... This is not the purpose of this anonymous tech and, furthermore, I think it works against it! 

I think we need to rethink about a low censoring tech, a way/methodology to reduce this garbage, unacceptable and punishable content.

A censor community methodology or something like this, that will protect eyes and mental health from these horrors since this tech is not legal and we could not report such content to authorized authority in order to attempt saving from these atrocities .

The engineer is not responsible for the way people are using of their tech, but if it can fix it but don't do it, he is responsible.

I don't want a censored web, this is unacceptable, but no censoring at all is unacceptable to since some contents are again humans rights and integrity.

So, please, discuss a low censoring procedure here, in community and in respect of everyone humans rights.

PS: Sorry for my poor English

**Comment by IGLOU-EU** [comment]
And this is not an April 1st Jokes ...

**Comment by latenightz** [comment]
I 100% agree with you. Freedom is great, but we need censorship too. I'm not saying we should censor important things like journalism and that stuff, but CP, r*pe and murder should NOT be allowed anywhere.

**Comment by deathtrip** [comment]
Your post is just a lot of bullshit, with no evidence. Everyone knows that it's not like when you open the Tor Browser or ZeroNet and immediately are bombarded with illegal content, like you want to make it seem.

If what you say is true and you really found all those things, then you must have actively searched for it.
Even law enforcement agencies have problems getting into these illegal communities you talk about. But you supposedly  "tested" something and found all of it right away.

I suggest you stop spreading FUD and go back to the big-tech controlled sites, hopefully they have enough censorship to suit your needs.

**Comment by rebelloV** [comment]
deathtrip, your logic is futile.

All you have to do on ZeroNet, is to go to any of the Zites offering public file hosting and sharing (one of which is indexed), and some files which contain harmful content similar to the aforementioned are blatantly listed as such, that’s why I personally no longer use this.

Of course, you don’t see all this immediately, but it’s easier to stumble across than you think. Are we to assume that we should let these things go unnoticed and just accept that they exist, and people may be harmed by them and accept it?

I personally think that would be a *very* dangerous course of action.

If you want this to get better, moderation is required, not censorship, they are two different things.

**Comment by deathtrip** [comment]
Just like the "moderation" on youtube, facebook, twitter etc.?  No thanks.
Both of you state that you don't use this network, yet you want to ruin it for everyone that does.

**Comment by gqgs** [comment]
>Are we to assume that we should let these things go unnoticed and just accept that they exist, and people may be harmed by them and accept it?

It's impossible to harm other by the simple act of sharing a sequence of 0s and 1s over the internet.

Nonetheless, there are plenty of sites in ZeroNet and outside of it with censorship/moderation.

Just use them if that's what you desire or just create and moderate your own.

**Comment by IGLOU-EU** [comment]
> > Are we to assume that we should let these things go unnoticed and just accept that they exist, and people may be harmed by them and accept it?
> It's impossible to harm other by the simple act of sharing a sequence of 0s and 1s over the internet.
> Nonetheless, there are plenty of sites in ZeroNet and outside of it with censorship/moderation.
> Just use them if that's what you desire or just create and moderate your own.

Your bank account is a sequence of 0s and 1s ...
The digital war is a sequence of 0s and 1s ...
Communication is a sequence of 0s and 1s ...
...

> Just like the "moderation" on youtube, facebook, twitter etc.? No thanks.
> Both of you state that you don't use this network, yet you want to ruin it for everyone that does.

You see someone speaking about or request GAFAM censoring ? I think not !

I only propose a reflection about `Methodology of censorship`, not a formal censorship like GAFAMs.
But a new way to limit this garbage content, a methodology that pays attention to freedom.

**Comment by IGLOU-EU** [comment]
I had thought of a community moderation by 'rating', like +/-, but a "review bomb" can be introduced and easily break the moderation ...

I look forward to your ideas and proposals. It is an interesting research for me and a BIG improvement of freedom to find a new method of moderation.

**Comment by rebelloV** [comment]
> It's impossible to harm other by the simple act of sharing a sequence of 0s and 1s over the internet.

On a *technical* level, you are correct, but we’re thinking about *content itself* here. What if someone, say, finds a picture which *mentally* scars them? In reality, there are practices to try and decrease the likelihood of this happening, but on an open network, there isn’t.

**Comment by rebelloV** [comment]
> Just like the "moderation" on youtube, facebook, twitter etc.? No thanks.
> Both of you state that you don't use this network, yet you want to ruin it for everyone that does.

I don’t even use any of the services listed above. I *formerly* used this network, and I would like to. How can we be ruining it? We are simply expressing our opinions on the matter to avoid people getting mentally scarred or hurt.

**Comment by gqgs** [comment]
>Your bank account is a sequence of 0s and 1s ...
>The digital war is a sequence of 0s and 1s ...
>Communication is a sequence of 0s and 1s ...

Irrelevant. If you think those can cause harm you should check the definition of the word "harm" again.

>In reality, there are practices to try and decrease the likelihood of this happening, but on an open network, there isn’t.

That's not true. If being _mentally scared_ is a concern there are block lists that you can follow to decrease the likelihood of seeing something offensive.

Or just use sites that only display content after it has been reviewed by a moderator.

Or just use sites that don't display anything other than text.

Those are things that can already be implemented by the people creating the sites. It doesn't have much else to do with ZeroNet itself.

**Comment by IGLOU-EU** [comment]
I think we get it, you don't want to participate constructively in this thread. If you ever want to contribute solutions or ideas, be free :(

**Comment by mSNAv9cYMZfkBn23** [comment]
> Of course, you don’t see all this immediately, but it’s easier to stumble across than you think. Are we to assume that we should let these things go unnoticed and just accept that they exist, and people may be harmed by them and accept it?
> 
> I personally think that would be a _very_ dangerous course of action.
> 
> If you want this to get better, moderation is required, not censorship, they are two different things.

Regarding actual illegal communities: did prohibition ever work? 
I think Zer0net has to implement a system that organises content based on the amount of active seeders it has. Logically speaking, very few users will seed CP and such. So this is one level of 'passive social moderation'. 

The next level of social moderation are the blocklists. Known users that grow to be trusted members of the community can manage blocklists. People are free to sign up to these if they want to. Zer0net also needs to implement this feature into the ZeroHello page. The individual blocklists will also be sorted according to popularity. Different blocklists can be made with different 'flavours'

The third and last level op moderation is personal moderation. If you don't want to see filthy shit, don't go looking for it. And if a gore poster invades your community's space, I'm sure the communal blocklists will quickly catch them. 

AT NO POINT SHOULD WE INTRODUCE A SINGLE FORM OF CENTRALISED POWER - ONCE THIS STEP IS TAKEN, YOU CANNOT GO BACK - IT WILL INEVITABLY LEAD TO CANCEL CULTURE - IN STEAD: LET THIS BE PLATFORM TO TEACH PEOPLE HOW TO SELF-MODERATE

**Comment by mSNAv9cYMZfkBn23** [comment]
>Regarding actual illegal communities: did prohibition ever work?

and let's not forget that the authorities use illegal content to bait criminals as well as subvert legit communities.

**Comment by IGLOU-EU** [comment]
@mSNAv9cYMZfkBn23 With your 11 days old account, you are probably a troll... But I'll answer anyway.

I think you must not have really read the exchanges. We have NEVER talked about a central censorship power, but a reflection on a new methodology

A particular point that you mention and was I like very much, is that indeed there is already a minimal sorting by the seeds. And this is a very, VERY good point !

I invite you to submit an idea of respect for individual freedom

**Comment by mSNAv9cYMZfkBn23** [comment]
@IGLOU-EU 
>troll

based on my serious, non-shitpost reply you think me a troll? Think moar. 
No one needs to be able to link my real GitHub account to Zeronet. You do know who own github?

**Comment by mSNAv9cYMZfkBn23** [comment]
@IGLOU-EU 
>already a minimal sorting by the seeds

Yeah but it seems to be based on total of seeds but not ACTIVE seeds. The number zeronet currently uses contains mostly dead seeds. I think it should exclusively live seeds. And a live seeds could be someone who is currently seeding or has seeded within the last 7 days (or whatever). 

Another idea is to give each site a RATING. The rating is based on how many times the site got blacklisted.

**Comment by mSNAv9cYMZfkBn23** [comment]
@IGLOU-EU 
one more final idea: create a SAFE BROWSING mode. This way you can visit a site, but every image and video is completely blurred out so that you can vaguely see what the content is, but not entirely. Only when you want to see something, you click on it and it becomes visible. 
I know it already exists in some form but it needs to be implemented better into the UX. Zeronet has quite a bit of a learning curve and we need to improve on this if we wish to grow the community.

**Comment by rebelloV** [comment]
> > Of course, you don’t see all this immediately, but it’s easier to stumble across than you think. Are we to assume that we should let these things go unnoticed and just accept that they exist, and people may be harmed by them and accept it?
> > I personally think that would be a _very_ dangerous course of action.
> > If you want this to get better, moderation is required, not censorship, they are two different things.
> 
> Regarding actual illegal communities: did prohibition ever work?
> I think Zer0net has to implement a system that organises content based on the amount of active seeders it has. Logically speaking, very few users will seed CP and such. So this is one level of 'passive social moderation'.
> 
> The next level of social moderation are the blocklists. Known users that grow to be trusted members of the community can manage blocklists. People are free to sign up to these if they want to. Zer0net also needs to implement this feature into the ZeroHello page. The individual blocklists will also be sorted according to popularity. Different blocklists can be made with different 'flavours'
> 
> The third and last level op moderation is personal moderation. If you don't want to see filthy shit, don't go looking for it. And if a gore poster invades your community's space, I'm sure the communal blocklists will quickly catch them.
> 
> AT NO POINT SHOULD WE INTRODUCE A SINGLE FORM OF CENTRALISED POWER - ONCE THIS STEP IS TAKEN, YOU CANNOT GO BACK - IT WILL INEVITABLY LEAD TO CANCEL CULTURE - IN STEAD: LET THIS BE PLATFORM TO TEACH PEOPLE HOW TO SELF-MODERATE

I personally believe *this* and the latter mentioned safe browsing mode would be the best course of action.

**Comment by IGLOU-EU** [comment]
> Yeah but it seems to be based on total of seeds but not ACTIVE seeds. The number zeronet currently uses contains mostly dead seeds. I think it should exclusively live seeds. And a live seeds could be someone who is currently seeding or has seeded within the last 7 days (or whatever).

It is a strange behavior, not to see the number of seeds currently available. Ticket ?

> Another idea is to give each site a RATING. The rating is based on how many times the site got blacklisted.

Yes, but RATING is very sensitive to bombing attack ...

---

I'm torn between wanting this content censored so that it's no longer online, or being able to be sure to avoid it.

After all, I don't like the fact that this kind of great technology makes life easier for the people who share, watch or realize this kind of content.

**Comment by mSNAv9cYMZfkBn23** [comment]
>After all, I don't like the fact that this kind of great technology makes life easier for the people who share, watch or realize this kind of content.

I think most everyone is but that's why we have the clearnet. This project is specifically meant to be a place where no rules apply in terms of censorship. If we would remove malicious content and violate another's choice to put those things online, that would be hypocritical, would it not? And again, I wish to draw your attention to the simple fact that prohibition never has and never will work. The illegality of it will only increase the attractiveness. 
The best we can do is simply provide every individual users with the best tools to have the ability to make their own choice in regards to what they wish to see and what they wish to avoid. 
This is why I'm not officially tying my name to this; because some people are not yet ready to understand why free speech is so important. 
And I believe that we are on a slippery slope if we decide to *censor* some things. Ideally, there would be no power levels. If there are no power levels, power cannot be abused. Because power does corrupt; that's the nature of it. It takes a virtually perfect human being to not abuse power or fall prey to its temptations. 

I truly understand your point of view, but if you enable centralised moderation it would be like opening Pandora's box: at some point the platform will become more popular, many more people will join but then you'll have to deal with hordes of people who demand a little bit more censorship. You think their request is reasonable, so you comply. But then after a while, they want even more censorship. You think their request is a bit exaggerated but you comply to keep everyone happy. Then some more time passes and all of a sudden people start to make the most ridiculous censorship requests. And by now you can't even decline anymore because the mob has gained political power and you're simply forced to comply and watch the platform being destroyed and there is nothing you can do about it. 

Like I said: for those who want censorship, the clearnet is sensible. For those who want absolute freedom of speech, a place like zeronet is more appropriate. And as long as those applications give the individual user the tools to filter content according to one's wishes, it is ethically acceptable.

Many people are killed by guns. But is it really the gun that kills?

**Comment by VitAndrGuid** [comment]
Please keep your cry for censorship out of techinical issues, thanks.

**Comment by Gobz** [comment]
So we're talking a zite that allows public posting? If that is the case then we simply let the zite add a moderation list users are notified about when they enter the zite. The list contains cryptographic signatures of trusted peers chosen as moderators, the moderators then add the ID's of the people posting illegal content to a block list signed with the moderator key, software running on the zite then keeps this list updated and automatically deletes the content from the perpetrating ID. This is of course optional, and communities can make their own moderation key lists if they disagree with the decisions of the current zite maintainers. Laws don't recognize whether you manually downloaded something illegal or if your browser did so a pretty strong argument can be made for moderation until laws are adapted to p2p(if ever).

**Comment by mSNAv9cYMZfkBn23** [comment]
@dergobi, when laws are adapted, that's when the anonymity layer has to be guaranteed. Perhaps it is easily solved by replacing the Tor layer with i2P. 
Another issue however, is that users can generate an infinite amount of ID's. Spammers would simply change ID's and evade the blocklists. Maintaining the lists will take a lot of work because eventually you'll be fighting bots if the platform ever takes off.

**Comment by Gobz** [comment]
@mSNAv9cYMZfkBn23 That is a problem yes. To counter bots ZeroNet needs rate limiting. Currently this is done on some zites through ZeroID but this introduces reliance on a centralized service for them to stay operational. I propose that zites are allowed to assign ID's as captcha providers(possibly the trusted moderators) then when you want a new identity, your machine sends a request to one of them and their machine then generates and verifies a captcha challenge. This would allow security on par with clearnet sites, though you wouldn't be able to rangeban addresses from datacenters normally used for spam but this is intentional. I agree on switching over to I2P as it is designed for P2P communication, where as the Tor network is actively harmed by such activity which leads to guard nodes blocking Zeronet users or throttling them with PoW challenges.

**Comment by gqgs** [comment]
>I propose that zites are allowed to assign ID's as captcha providers(possibly the trusted moderators) then when you want a new identity, your machine sends a request to one of them and their machine then generates and verifies a captcha challenge.

You can already do something like that. Here's how a decentralized captcha system could work.
```json
 "user_contents": {
  "cert_signers": {
   "request": ["..."],
   "approved": ["..."]
  },
```
`request` and `approved` are the only domains allowed to post in the site.
`request` certs are self-signed while `approved` certs are signed by the site owner or trusted moderators.

```json
  "permission_rules": {
   ".*@request": {
    "files_allowed": "captcha.json",
    "max_size": 5000
   },
   ".*@approved": {
    "files_allowed": "data.json",
    "files_allowed_optional": "...",
    "max_size": 1000000
   }
```
`@request` users can only publish content related with the captcha while `@approved` users are allowed to fully interact with the site.

The process of acquiring a signed `approved` certificate is as follows:
1. The user creates a self-signed `request` certificate
2. The user publishes in `captcha.json` requesting a new captcha challenge to be solved
3. The site owner or one of the trusted moderators see the request then generate and publish a new captcha challenge for that specific user
4. The user receives the challenge and submits a solution
5. After receiving the answer the person who generated the challenge can either deny the request, possibly because of an incorrect solution or an expiration time, or return a valid signed `approved` certificate
6. At this point the user is either free to post on the site with the signed `approved` certificate or go back to (2) if the request was denied

To revoke the permission for any user for whatever reason the `permissions` object can be used.

The only issue I noticed with this architecture is that you cannot remove a specific moderator without also invalidating all certificates that were signed by him.

**Comment by mSNAv9cYMZfkBn23** [comment]
@gqgs 
>site owner/moderator certificate signing

This applies a centralised methodology. I would agree that you enable the feature in ZeroNet but also offer a decentralised alternative. The latter could be an invite system: users who already own a zite certificate are able to invite others. They way this could play at is as follows: 

1. new user arrives on zite
2. new user requests zite ID (zites can in turn request the user to type out a short motivation letter and/or answer some questions) 
3. any existing user can grant access to the new ID (but perhaps it would be wise to offer the zite owner to set a quota - for example: one user can grant a maximum of 5 ID's per week)

**Comment by gqgs** [comment]
@mSNAv9cYMZfkBn23 
>This applies a centralised methodology

How exactly is multiple entities being able to grant write permission to the site an application of a centralized methodology?

>zites can in turn request the user to type out a short motivation letter and/or answer some questions
>any existing user can grant access to the new ID

That sounds quite prone to abuse.

**Comment by mSNAv9cYMZfkBn23** [comment]
@gqgs 
I'm not sure if you are aware, but the whole point of zeronet is decentralisation. If you want a perfect PC internet experience, you can get that on the clearnet. 

>How exactly ...

You said the site owner or one of the trusted moderators 

>can either deny the request, possibly because of an incorrect solution or an expiration time, or return a valid signed approved certificate

This means that only the group that (or individual) that holds the power position, decides who can join and who can not. 

>That sounds quite prone to abuse.

of course; but it is up to the community (not an individual) to establish and enforce the appropriate rules for the zite. 

The only thing we would like to prevent, is that communities are defenceless (powerless) against exogenous threats like spammers, bots, glowies and raids from other communities. In a way you want to encourage the no-censorship ideology, but you also want to give them the tools to establish something that you could describe as a fusion of anarchy and a republic. Anarchy preserves free speech and the republic protects the culture or identity of the community from external threats. It's always a balance between freedom and control. On the clearnet there is little freedom and much control. On zeronet there is a lot of freedom and little control; and the little control there is, is as decentralised as possible and requires community engagement in order to be effective. Strong, healthy communities will be able to use these tools to fend off threats. Weak, unhealthy communities will be more prone to disintegration.

**Comment by gqgs** [comment]
@mSNAv9cYMZfkBn23 
>I'm not sure if you are aware, but the whole point of zeronet is decentralisation

I have absolutely no clue what gave you the impression I think otherwise.

>This means that only the group that (or individual) that holds the power position, decides who can join and who can not.

That in no way, shape or form implies centralization. There would be nothing preventing you from making every single user in the site a "trusted moderator" if that's what you wish.
If want every single member to have the same "power" then that's a different issue that would require a redesign of ZeroNet itself since the site owner will always have more "power" than regular users.

>of course; but it is up to the community (not an individual) to establish and enforce the appropriate rules for the zite.

I think you misunderstood the point.
Your system where everyone can create accounts by inviting each other can be easily exploited to create an exponential number of accounts which can then be used to overpower the system.
It would be unfeasible to do a similar attack in the system I mentioned above.

**Comment by mSNAv9cYMZfkBn23** [comment]
>no clue what gave you the impression I think otherwise

Apologies. It seems to me you haven't fully grasped the scope of decentralisation. 

>There would be nothing preventing you from making every single user in the site a "trusted moderator" 

Who has this authority though? Herein lies the centralisation. 

>That in no way, shape or form implies centralization

Yes it does. Let's take a look at the dictionary, shall we? https://www.merriam-webster.com/dictionary/centralization

>2: to concentrate by placing **power and authority in a center or central organization**

>would require a redesign of ZeroNet itself since the site owner will always have more "power" than regular users

Yes, you're right. 

>Your system where everyone can create accounts by inviting each other can be easily exploited to create an exponential number of accounts which can then be used to overpower the system.
It would be unfeasible to do a similar attack in the system I mentioned above.

Correct. But like I said in the previous message: the power structure in any society is always a balancing of freedom and control. In your system, there is very little freedom because mods can easily decide to remove or withhold people with different ideologies from entering the zite. Sure this can be used for the good of community, but in another's eyes this is precisely what cencorship is. Herein lies the paradox and this issue is as old as human kind itself. Essentially, every system can be both good or evil in the eyes of different individuals. Every system can work harmoniously or disharmoniously. It depends on the individuals who are in power, as well as the state of being of all of the individuals who make up the society. It's a complex matter and much wiser people than the of us have written about these irreconcilable conundrums.  

To conclude: Zeronet is an alternative to the clearweb. It is not meant to run the same clean and controlled societies we find on the clearweb. Rules and morality, by intention, has to be somewhat vague on Zeronet. The individual can protect himself from this vagueness in two ways: 
1. don't even bother going to Zeronet
2. use self-moderation

**Comment by mSNAv9cYMZfkBn23** [comment]
@gqgs 
to make a long story short, I would argue that our goal is to strip the power structure altogether, but in turn offer tools to the individual (and to lesser extent a community) to protect himself from abuse. 

And what is abuse defined in this context? 
1. spam
2. accidentally seeding illegal content

These are the only problems we need to solve: first make it impossible for bots to spam and render places unusable and second, to redesign the UX and GUI so that accidentally seeing and hosting illegal content becomes almost impossible while still leaving the choice to the individual. By default you encourage communities to self-moderate illegal content (for their own good) but you still leave them the option to choose to not self-moderate. You leave the choice up to the individual as well as the community.

**Comment by rebelloV** [comment]
⬆️

Above, I believe that @mSNAv9cYMZfkBn23 has got this right, moving towards the aforementioned ‘redesign’ would ruin the point of Zeronet.

We need and require better self-moderation tools for the individual, a previously mentioned UI design to blur photos, would be incredibly useful.

Some block lists for certain zites possibly harbouring illegal content would be excellent, I don’t *think* these exist at the moment.

and of course, general social moderation via a cooperative and helpful community would be massively helpful as well (e.g. Avoiding seeding zites that harbour generally illegal content, informing other users of possible spam etc.)

Overall, I think the only way to fully put my concerns to rest would be to change U.K. and US law to take into account how P2P works and how they get someone into trouble unknowingly, but, I can’t do that, so... 🤷‍♂️

**Comment by gqgs** [comment]
@mSNAv9cYMZfkBn23 

>It seems to me you haven't fully grasped the scope of decentralisation.

I think the issue is me having to grasp your understanding of the concept.
Apparently the misunderstanding we're having is because you seem to think you cannot have hierarchy in a decentralized system.

>2: to concentrate by placing power and authority in a center or central organization

If multiple people have the power and authority to do something that's quite literally the opposite of concentrating the power and authority in centralized manner.

Since you like definitions you should check this one:

https://www.merriam-webster.com/dictionary/decentralization

>1. : the dispersion or distribution of functions and powers

Nothing in this definition implies that every part of the decentralized system needs to have the same function or power.

>Who has this authority though? Herein lies the centralisation.

That's specific to the implementation.
Your assumption that it has be to be a centralized authority is erroneous and shortsighted.
Essentially the only thing you need to accomplish this in a decentralized manner is a consensus algorithm.
If you read the literature you'll see plenty of ways this can accomplished.

>Yes, you're right.

I'm glad we agreed on something.
It sounds this should be proposed in another issue though since it doesn't have much to do with the original post anymore.

**Comment by mSNAv9cYMZfkBn23** [comment]
@rebelloV 
>I don’t think these exist at the moment.

this is a test blocklist someone made: http://127.0.0.1:43110/16soWyYc1kTobaevbN25FJjWL2qf9PWvFG/ 

>to take into account how P2P works and how they get someone into trouble unknowingly

You are a conscientious person and it graces you but keep in mind that zeronet is just a tool and it's not your responsibility if other people abuse this tool. But what is important is that you provide clear instructions on how to use the tool correctly. Certain zites already implement warning messages and I think it is a good idea to expand on this too. And we also need easily obtainable tutorials that explain the risks of zeronet so that people have gotten all the information they need to make an informed choice. 

@gqgs 
>is because you seem to think you cannot have hierarchy in a decentralized system.

In a sense decentralisation is just a spectrum, I agree with that. But I never said hierarchy doesn't belong in decentralised systems. What I said was that there is always the question of balancing freedom (sovereignty) vs control (hierarchy). 

>Nothing in this definition implies that every part of the decentralized system needs to have the same function or power.

The reason why this hasn't really been tried in the real world is because it would be impossible to manage; it is the same issue with a 'direct democracy'. But a community run on the internet has the ideal infrastructure to provide the opportunity to implement this system. 

If you do not completely decentralise your system, it is still vulnerable at the nerve centers. Moderators can be influenced and corrupted. Even the zite owner can unwittingly recruit spy moderators. This happened recently on an fchan instance: a mod went crazy and cleared the entire board. True story. Another anecdotal yet important piece of evidence is 4chan. In 2008 its administrator was approached by the feds and forced to cooperate in a pedo sting operation. 
Check these image for evidence: 
- https://ibb.co/dLmhbDq
- https://ibb.co/b6w49gV
- https://ibb.co/xqfn4M8

What I propose is absolute decentralisation (anarchy) from a systemic point of view; but offer the users tools so that they can self-moderate and create their own communities if they wish to do so. Please try to understand why I am arguing this. Your proposition of dispersing power and functions is LITERALLY how almost every system on the clearweb works. And we all know what the clearweb is like. Zeronet is by definition the opposite of that. Why would you want to compromise that? Honestly, if you keep resisting reason, I'm going to assume you a glowie. 

>Essentially the only thing you need to accomplish this in a decentralized manner is a consensus algorithm.

Are you for real? How can you be so fallible? You literally just ignored the whole discussion. The whole point of this project is to remove any form of systemic consensus because this the ultimate tool for suppression and censorship. I thought you understood this? Am I talking to a wall?

**Comment by gqgs** [comment]
@mSNAv9cYMZfkBn23 

>Zeronet is by definition the opposite of that.
>The whole point of this project is to remove any form of systemic consensus because this the ultimate tool for suppression and censorship

That's your idealized view of the project.
In realitity, there are plenty of ways to moderate content in ZeroNet.
Site owners can moderate content in their sites.
Hub owners can moderate content in merger sites.
Some people would like those to remain the case.

If you disagree and think only self-moderation should exist that's fine but, as I mentioned before, that's a different issue that will most likely require you to fork ZeroNet to order to accomplish it.
I wish you good luck if you decide to pursue that.

**Comment by ghost** [comment]
You can create an adblock extension on top of your browser.

**Comment by mx5kevin** [comment]
To filter out harmful users like: political elements, government, who scan the network, they want to stop content because they just don’t like it. Nor are site owners interested in coming from certain users who want to say or force tham what they can and cannot publish. Especially against the parasitic  political elite need a strong global social power. Where citizens can kick the asses of the government and politicans. Nor does it work that the visitor wants free content, does not want to seed it, wants to control what other users can see. And content producers are not just not making any money, but they s pay to have their content online.

And a big problem all over the internet. It is not the content owner who has to pay to have their content online. The visitors must seed or pay for the content. The entire download system does not work. Antivirus softwares and this blocklists make it difficult on the network the crypto mining like monero mining using HTML with websites. Setting up larger websites is also inconvenient. Website and content owners need to filter out harmful visitors and bad peers.

Like bit torrent coin system, Filecoin, Storj. Maid Safe It would be useful a coin system where the users get money if seed the content. And pay money if they download content. Those who do not help the network with seeding correctly the content pay for it. And also gives a option where the visitors can pay for a paid hosting service, and money if the content owner setup if his site are paid. Content owners need to make money with their work.
Like this way:

1.setup: -If the user upload static are are less then dowload statics then with seeding, proof of work, or with money must pay for more downloads. If the user delete a site the user paid money or seedig to the content owner.

2.setup: -The site owners can setup when the user load a site or a file  a sum of money to allow users to access the content. 

3.Setup: -The content owner can use the revenues auto way the part of them to use it with a decentralized paid hosting system.

The ZeroNet project requires several independent development teams. The project is stalling in one place and there is no development. There is more progress in censorship than file sharing works correctly in the network, and this is the problem.

**Comment by mx5kevin** [comment]
With these bocklists there is a lot of abuse. All kinds of harmful elements make such a block list. The option is used for political purposes. There used to be a revenue opportunity for content owners with monero mining. The Zeronet blocklists and antivirus companies this option was ruined. Most block lists are operated by users who want to destroy this network. And they also want to infiltrate project development. Political interest wants to put pressure on content owners and this project developpers with all their might. The only working option is for everyone to post and share what they want. Those who want censorship here they must be persecuted from this project.

**Comment by mSNAv9cYMZfkBn23** [comment]
@mx5kevin 
>The only working option is for everyone to post and share what they want

honestly I think this is the best solution. Self-responsibility. If you can't handle that, you shouldn't be on zeronet

**Comment by freemotion21** [comment]
I think that Tor isn't bad, but the pages that we can find in it, maybe are very dangerous.
That's my opinion.

**Comment by styromaniac** [comment]
I've long-proposed a web of trust where you can rate a site, but your ratings mean nothing to other users unless they follow/subscribe to your reviews.

It's a very simple approach without encroaching upon any centralized censorship.

Say I've found bad content on a site, I down-vote the site, then you subscribe to my site reviews. You'll then be warned of sites that I've deemed unsafe.

This would be a more actively used feature than blocklists, thus you don't have to wait on someone like me who has the technical skills to maintain a blocklist like MOAB.

Censorship should always be opt-in, not involuntarily thrusted upon everyone by default.

**Comment by rebelloV** [comment]
@styromaniac This is ideal, is it possible this system can be developed and implemented?

**Comment by styromaniac** [comment]
> @styromaniac This is ideal, is it possible this system can be developed and implemented?

I think so, but I'm not a programmer.

**Comment by hardBSDk** [comment]
@IGLOU-EU Your logic applies for Tor too, you can't censor the websites you don't like there.

About ZeroNet, you are right on some parts since ZeroNet is P2P sites where everyone hosts a piece of the websites.

I recommend you to not judge the technology by its content, you are free to delete the websites you don't like and delete the files downloaded (it's ilegal, after all).
